modules/vye/spec/sidekiq/vye/sundown_sweep_spec.rb

```diff
@@ -19,4 +19,64 @@
     expect(Vye::SundownSweep::DeleteProcessedS3Files).to have_enqueued_sidekiq_job
     expect(Vye::SundownSweep::PurgesStaleVerifications).to have_enqueued_sidekiq_job
   end
+
+  # Exceptions are tested one by one rather than all at once because it's easier to debug
+  # if one of them is broken rather than trying to figure out which one in the loop is broken.
+  describe 'exception handling' do
+    let(:s3_client) { Aws::S3::Client.new(stub_responses: true) }
+    let(:logger) { Rails.logger }
+
+    before do
+      allow(Aws::S3::Client).to receive(:new).and_return(s3_client)
+      allow(logger).to receive(:error)
+    end
+
+    it 'throws an exception when the bucket cannot be found on AWS' do
+      s3_client.stub_responses(:delete_object, 'NoSuchBucket')
```

Comment:
I pulled it down to confirm, but these `stub_responses` lines aren't needed. The following line is already saying that "whenever `s3_client.delete_object` is called, raise ERROR". So I think you can just remove those 4 statements

Reply:
Let me take a look.

Reply:
That was correct and I removed them and pushed. Thx for the knowledge drop.